openpype/plugins/actions/open_file_explorer.py

```diff
@@ -0,0 +1,167 @@
+import os
+import platform
+import subprocess
+import getpass
+
+from openpype.client import (
+    get_project,
+    get_asset_by_name,
+)
+from openpype.pipeline import (
+    Anatomy,
+    LauncherAction,
+)
+
+
+class OpenTaskPath(LauncherAction):
+    name = "open_task_path"
+    label = "Explore here"
+    icon = "folder-open"
+    order = 500
+
+    def is_compatible(self, session):
+        """Return whether the action is compatible with the session"""
+        return "AVALON_ASSET" in session
+
+    def process(self, session, **kwargs):
+        from qtpy import QtCore, QtWidgets
+
+        project_name = session["AVALON_PROJECT"]
+        asset_name = session.get("AVALON_ASSET", None)
+        task_name = session.get("AVALON_TASK", None)
+        host_name = None    # never present in session
+
+        project = get_project(project_name)
+        asset = get_asset_by_name(project_name, asset_name)
+
+        data = self.get_workdir_data(project_doc=project,
+                                     asset_doc=asset,
+                                     task_name=task_name,
+                                     host_name=host_name)
+        anatomy = Anatomy(project_name)
+        result = anatomy.format(data)
+
+        # todo: implement custom template keys instead of 'work'
+        # Get template key
+        # template_key = get_workfile_template_key_from_context(
+        #    asset_name, task_name, host_name, project_name, dbcon
+        # )
+        workdir = result["work"]["folder"]
+
+        # Keep only the part of the path that was formatted by splitting up to
+        # the first stub - we can only explore up to there
+        STUB = "<<NULL>>"
+        valid_dir = workdir.split(STUB, 1)[0]
+
+        # If the path endswith `/work` and the path does not exist but the
+        # parent does then we allow to go to the parent because it could for
+        # example be the hierarchical folder for `asset` in e.g. `asset/hero`
+        if not os.path.exists(valid_dir) and valid_dir.endswith("/work/"):
+            # /folder/ dirname is /folder so we rstrip the last / to be sure
+            parent_dir = os.path.dirname(valid_dir.rstrip("/"))
+            if os.path.exists(parent_dir):
+                valid_dir = parent_dir
+
+        path = os.path.normpath(valid_dir)
```

Comment:
```suggestion
        # Path is not filled
        if not valid_path:
            return

        # Normalize
        valid_workdir = os.path.normpath(valid_workdir)
        while not os.path.exists(valid_workdir):
            prev_workdir = valid_workdir
            valid_workdir = os.path.dirname(prev_workdir)
            if not valid_workdir or valid_workdir == prev_workdir:
                return None

        path = valid_workdir 
```

Reply:
Even though I understand the logic - it's not entirely accurate to the artist. This could mean that clicking "Explore here" on a task or asset that hasn't been initialized that it'll still "think" it found a path even if the project folder itself doesn't even exist on disk. 

With this expected path:
`/projects/my_project/s01_e01/ep01/sh010/work/modeling` 

This could even just explore to `/projects` without any warning which to me personally feels misleading - it's nowhere near close to what the user would expect.

Reply:
Marking this unresolved because I feel my last comment wasn't addressed?

Reply:
Expecting `/work/` at the end is not expected either. Especially if is used only the part of template which is formatted (this is the first part which is clunky). Both don't make sense in the same way. Using `/work/` is more matching a specific work template.

Reply:
Can we limit it so it never passes the first `{project}` variable? 

Not entirely what we'd want. We sometimes also use the button to just quickly see if there were already ANY files whatsoever -then clicking it would say "Folder does not exist" and you'd know what's up. Now you'd be thrown into the folder regardless and need to look where you are to see whether the folder actually existed.

![image](https://github.com/ynput/OpenPype/assets/2439881/80d76a57-7f2d-4493-b291-e334157a1900)

Maybe we should allow it go up only to `{asset}`? (Even though technically a template could be: `{task}/{asset}`, maniacs!) Would it at least a sensible default?